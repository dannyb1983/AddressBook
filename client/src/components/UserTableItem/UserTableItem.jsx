//modules
import React, { Fragment, useState } from "react";
import { Link, useHistory } from "react-router-dom";
import { Image, Container, Row, Col, Tooltip, OverlayTrigger, ListGroupItem } from "react-bootstrap";

const UserTableItem = ({ people, setContactToEdit, searchInput }) => {
	const history = useHistory();
	// Tooltip
	const renderTooltip = (props) => (
		<Tooltip id="button-tooltip" {...props}>
			Select contact to show details
		</Tooltip>
	);
	// Search filter
	const filteredUsers = (userInput) => {
		if (!people) return;
		const lowerCaseUserinput = userInput.toLowerCase();
		let filteredResults = people.filter((person) => {
			return person.name.first.toLowerCase().includes(lowerCaseUserinput) || person.name.last.toLowerCase().includes(lowerCaseUserinput);
		});
		return filteredResults;
	};
	const filteredPeople = filteredUsers(searchInput);
	return (
		<Fragment>
			{people ? (
				filteredPeople.map((user, i) => {
					return (
						<ListGroupItem key={`${user.login.uuid} ${i}`}>
							<Container className="my-3 my-auto flex justify-content-center">
								<Row
									onClick={(e) => {
										setContactToEdit(user);
										history.push("/details");
									}}>
									<Col md={6} className="mx-auto text-center">
										<div className="ms-2 me-auto">
											<div className="fw-bold">
												<OverlayTrigger placement="right" delay={{ show: 250, hide: 400 }} overlay={renderTooltip}>
													<Link
														to="/details"
														style={{
															color: "inherit",
															textDecoration: "inherit",
															paddingRight: "20px"
														}}>
														{" "}
														<div className="my-auto">
															{user.name.first} {user.name.last}{" "}
														</div>
													</Link>
												</OverlayTrigger>
											</div>
										</div>
									</Col>
									<Col md={6} className="my-auto mx-auto text-center py-2 px-1">
										{/*User photos are here */}
										<Image height="48px" width="48px" src={user.picture.thumbnail || "../../../../assets/noimage.jpg"} alt="user_picture" roundedCircle />
									</Col>
								</Row>
							</Container>
						</ListGroupItem>
					);
				})
			) : (
				<div>Searching for users...</div>
			)}
		</Fragment>
	);
};

export default UserTableItem;
