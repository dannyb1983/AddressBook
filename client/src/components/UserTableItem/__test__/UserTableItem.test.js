//modules
import React from "react";
import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom"
import { contact, contacts } from "../../../../../test/jest/__mocks__/testData"
//components
import UserTableItem from "../UserTableItem";

describe('User Table Item (Unit)', () => {
    const renderUserTableItem = () => render(<MemoryRouter><UserTableItem people={contacts} setContactToEdit={contact} searchInput="" /></MemoryRouter>)
    const renderEmptyUserTableItem = () => render(<MemoryRouter><UserTableItem /></MemoryRouter>)

    describe('render()', () => {
        const { asFragment } = renderUserTableItem();
        it('should match snapshot', () => {
            expect(asFragment()).toMatchSnapshot();
        });
    });
    
    describe('User Table edge cases + undefined values', () => {
        it('should render an error message when given no input values', () => {
            renderEmptyUserTableItem()
            expect(screen.getByText("Searching for users...")).toBeInTheDocument();
        });
    });
});