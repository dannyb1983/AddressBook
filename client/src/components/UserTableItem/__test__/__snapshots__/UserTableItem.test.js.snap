// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Table Item (Unit) render() should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="list-group-item"
  >
    <div
      class="my-3 my-auto flex justify-content-center container"
    >
      <div
        class="row"
      >
        <div
          class="mx-auto text-center col-md-6"
        >
          <div
            class="ms-2 me-auto"
          >
            <div
              class="fw-bold"
            >
              <a
                href="/details"
                style="text-decoration: inherit; padding-right: 20px;"
              >
                 
                <div
                  class="my-auto"
                >
                  John Smith 
                </div>
              </a>
            </div>
          </div>
        </div>
        <div
          class="my-auto mx-auto text-center py-2 px-1 col-md-6"
        >
          <img
            alt="user_picture"
            class="rounded-circle"
            height="48px"
            src="https://randomuser.me/api/portraits/thumb/men/33.jpg"
            width="48px"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    class="list-group-item"
  >
    <div
      class="my-3 my-auto flex justify-content-center container"
    >
      <div
        class="row"
      >
        <div
          class="mx-auto text-center col-md-6"
        >
          <div
            class="ms-2 me-auto"
          >
            <div
              class="fw-bold"
            >
              <a
                href="/details"
                style="text-decoration: inherit; padding-right: 20px;"
              >
                 
                <div
                  class="my-auto"
                >
                  Carter Robinso 
                </div>
              </a>
            </div>
          </div>
        </div>
        <div
          class="my-auto mx-auto text-center py-2 px-1 col-md-6"
        >
          <img
            alt="user_picture"
            class="rounded-circle"
            height="48px"
            src="https://randomuser.me/api/portraits/thumb/men/94.jpg"
            width="48px"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    class="list-group-item"
  >
    <div
      class="my-3 my-auto flex justify-content-center container"
    >
      <div
        class="row"
      >
        <div
          class="mx-auto text-center col-md-6"
        >
          <div
            class="ms-2 me-auto"
          >
            <div
              class="fw-bold"
            >
              <a
                href="/details"
                style="text-decoration: inherit; padding-right: 20px;"
              >
                 
                <div
                  class="my-auto"
                >
                  Carter Robinson 
                </div>
              </a>
            </div>
          </div>
        </div>
        <div
          class="my-auto mx-auto text-center py-2 px-1 col-md-6"
        >
          <img
            alt="user_picture"
            class="rounded-circle"
            height="48px"
            src="https://randomuser.me/api/portraits/thumb/men/94.jpg"
            width="48px"
          />
        </div>
      </div>
    </div>
  </div>
  <div
    class="list-group-item"
  >
    <div
      class="my-3 my-auto flex justify-content-center container"
    >
      <div
        class="row"
      >
        <div
          class="mx-auto text-center col-md-6"
        >
          <div
            class="ms-2 me-auto"
          >
            <div
              class="fw-bold"
            >
              <a
                href="/details"
                style="text-decoration: inherit; padding-right: 20px;"
              >
                 
                <div
                  class="my-auto"
                >
                  Sandyw Scholl 
                </div>
              </a>
            </div>
          </div>
        </div>
        <div
          class="my-auto mx-auto text-center py-2 px-1 col-md-6"
        >
          <img
            alt="user_picture"
            class="rounded-circle"
            height="48px"
            src="https://randomuser.me/api/portraits/thumb/women/58.jpg"
            width="48px"
          />
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
